allow override of /etc/diag.sh by sourcing an /etc/diag.arch.sh, if existing

diff --git a/target/linux/lantiq/base-files/etc/diag.sh b/target/linux/lantiq/base-files/etc/diag.sh
--- a/target/linux/lantiq/base-files/etc/diag.sh
+++ b/target/linux/lantiq/base-files/etc/diag.sh
@@ -40,3 +40,7 @@ set_state() {
 		;;
 	esac
 }
+
+if [ -e /etc/diag.arch.sh ]; then
+	. /etc/diag.arch.sh
+fi
